### PR TITLE
Doc updates for new user permissions

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/945

Adding what I have as a base PR for the changes coming August 8th, some other docs might also need to be updated. For example, I'm on the fence about whether we add a screenshot of the new "new Organization member" page to `add-user.md`. Would that screenshot help users?

Release notes should go in a separate PR alongside all other release notes for that week.

### CODE REVIEW

**Review by kwcanuck — APPROVED** [review]

**Comment by chomipi88** [comment]
@jwitz @kwcanuck  I do believe we should document the updated ways and experience of getting access to Astro.

1. A new user is invited to an Organization or logs in via an Okta or SAML connection to their Organization. They will see the below Org Home Page view:

image

2. Without being invited, a new user logs in via Basic Auth, Google, or GitHub using an email domain that is configured for the Organization. They will see the below Org Home Page view:

image

**Comment by kwcanuck** [comment]
@chomipi88 I think your concerns are addressed in the existing topic _[Log in to Astro](https://docs.astronomer.io/astro/log-in-to-astro)_. As you can see, we discuss IdP access and additional access options. If you think additional information is required to clarify the behavior, I can add text similar to the following:

_If your Organization is using an Identity Provider (IdP) or a SAML connection for user authentication, you'll receive an email inviting you to the Organization Workspace. Follow the instructions in the email to access the Cloud UI and your Workspace. If your Organization is not using an IdP or SAML connection for user authentication, see Log in to the Cloud UI._

In my opinion, I don't think this text is necessary as the information is clear as is.

File: astro/user-permissions.md [hunk 1]
```diff
@@ -37,20 +41,20 @@ The first 3 users that log in to a new Organization automatically become Organiz
...
+| View DAGs in the Airflow UI              | ✔️                   | ✔️                   | ✔️                  |
+| View Airflow task logs                   | ✔️                   | ✔️                   | ✔️                  |
+| View Airflow connections and Variables   |                      |                      | ✔️                  |
+| Update user roles and permissions        |                      |                      | ✔️                  |
+| Update Airflow connections and Variables |                      |                      | ✔️                  |
+| Update Deployment configurations         |                      | ✔️                   | ✔️                  |
+| Edit task instance and DAG state         |                      | ✔️                   | ✔️                  |
+| Push code to Deployments                 |                      | ✔️                   | ✔️                  |
+| Invite users to a Workspace              |                      |                      | ✔️                  |
+| Delete a Deployment                      |                      |                      | ✔️                  |
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/user-permissions.md:57]: @chomipi88 I updated the permissions for deleting a Deployment. I didn't see an entry for creating a Deployment.

File: astro/user-permissions.md [hunk 2]
```diff
@@ -11,13 +11,17 @@ To better protect your data pipelines and cloud infrastructure, Astro offers rol
...
 
+## Organization Roles
+
+An Organization role grants a user some level of access to an Astro Organization, including all of the Workspaces within that Organization. All users have an Organization role regardless of whether they belong to a Workspace. The following Organization roles are available:
+
 | Permission                                                  | **Organization Member** | **Organization Billing Admin** | **Organization Owner** |
 | ----------------------------------------------------------- | ----------------------- | ------------------------------ | ---------------------- |
-| Create a new Workspace                                      | ✔️                      | ✔️                             | ✔️                       |
-| View Organization details and user membership               | ✔️                      | ✔️                             | ✔️                       |
-| View lineage data in the **Lineage** tab                    | ✔️                      | ✔️                             | ✔️                       |
```
[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/user-permissions.md:18]: @chomipi88 unfortunately @jwitz edited this table and I'm not sure what he had in mind. To me, viewing Lineage data is different than viewing Organization details and user memberships.

File: astro/user-permissions.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (chomipi88) [reply astro/user-permissions.md]: Let's add an entry for Creating a Deployment or group it with Delete

[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/user-permissions.md]: @chomipi88 the table has been updated.

File: astro/user-permissions.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (chomipi88) [reply astro/user-permissions.md:18]: @kwcanuck I agree. Its a significant part of the product and we should call it out separately.
Let's add it back. To confirm, all Org roles can view Lineage.

[the remark this answers is not in the document]
Comment (kwcanuck) [reply astro/user-permissions.md:18]: @chomipi88 I have revised the permissions table as requested.

**Pull request merged**